backend/src/socket/socket.gateway.ts

```diff
@@ -34,242 +36,186 @@ export class SocketGateway implements OnGatewayConnection {
-    if (!playerEmail) return;
-    await this.cacheManager.del(client.id);
-
-    const existingPlayer: string = await this.cacheManager.get(
-      `game${playerEmail}`,
-    );
-    if (!existingPlayer) return;
-    const existingPlayerObject: Player = JSON.parse(existingPlayer);
-
-    // find the user in database
-    const user = await this.prisma.user.findUnique({
-      where: {
-        email: existingPlayerObject.email,
-      },
-    });
-    // check the user status in database;
-    if (!user) {
-      console.log('user not found!');
-      return;
-    }
-    if (user.status === 'OFFLINE') {
-      this.cacheManager.del(`game${existingPlayerObject.email}`);
-      return;
-    } else if (user.status === 'WAITING') {
-      const pendingPlayer: string = await this.cacheManager.get('pendingPlayer');
-      if (!pendingPlayer) return;
-      // check if pendingPlayer(in cache) is the waiting player(in database)
-      const pendingPlayerObject: Player = JSON.parse(pendingPlayer);
-      if (user.email === pendingPlayerObject.email) {
-        await this.cacheManager.del('pendingPlayer');
-        await this.cacheManager.del(`game${user.email}`);
-      }
-    } else if (user.status === 'PLAYING') {
-      // wait 30 seconds, or game is lost??
-      if (existingPlayerObject.gameID) {
-        client.join(existingPlayerObject.gameID.toString());
-      }
-    }
-    console.log(`Socket: ${client.id} disconnected from game socket. 0.0`);
+    await this.gameService.cancelPendingGame(client);
+    await this.gameService.updateUserDisconnectStatus(client);
+    await this.gameService.clearData(client);
+    console.log(`\x1b[31m Socket: ${client.id} disconnect from Game Socket! \x1b[0m`);
   }
 
   /****************************************************************************/
-  /* GAME                                                                     */
+  // Find game
   /****************************************************************************/
-  @SubscribeMessage('setCanvas')
-  handleSetCanvas(client: Socket, payload: any) {
-    this.gameService.setCanvas(payload);
-  }
-
   @SubscribeMessage('findGame')
   async handleFindGame(client: Socket) {
-    const currentPlayerEmail: string = await this.cacheManager.get(client.id);
-    const currentPlayer: string = await this.cacheManager.get(
-      `game${currentPlayerEmail}`,
-    );
+    let currentPlayer = await this.gameService.createPlayer(client);
     if (!currentPlayer) return;
-    const currentPlayerObject: Player = JSON.parse(currentPlayer);
 
-    const startGame: [boolean, string] =
-      await this.gameService.joinOrCreateGame(currentPlayerObject);
-    const gameRoom: string = currentPlayerObject.gameID.toString();
+    const pausedGameID = await this.gameService.findPausedGame(client);
+    if (pausedGameID) 
+      currentPlayer.gameID = pausedGameID;
 
+    let newGame: Game = await this.gameService.createGame(client);
```

Comment:
You can pass 'currentPlayer' here in stead of 'client', so you don't have to do the cache calls again.

Reply:
This is double: `createGame()` is executed, even when a paused game is found. And subsequently, in create game, the function `findPausedGame` is executed once more.

Reply:
I'm kinda doing a double check so I don't need to add more condition in socket.gateway. Since things are  working now I'll try to remove the repetition latter. 😅